File: android/src/main/java/com/azifacemobile/theme/AdditionalReview.java
package com.azifacemobile.theme;

import com.azifacemobile.theme.abstracts.ViewStyle;
import com.azifacemobile.utils.Theme;
import com.facebook.react.bridge.ReactApplicationContext;

import org.json.JSONException;
import org.json.JSONObject;

public class AdditionalReview extends ViewStyle {
  private static final String KEY = "additionalReview";
  final Theme theme;
  final JSONObject target;
  final Color color;
  final Image image;

  public AdditionalReview(ReactApplicationContext context, JSONObject target) {
    super(target, KEY);

    this.theme = new Theme();

    this.target = this.theme.getTarget(target, KEY);
    this.color = new Color();
    this.image = new Image(context, this.target);
  }

  private boolean getBoolean(String key, boolean defaultValue) {
    try {
      if (!this.theme.exists(this.target, key)) {
        return defaultValue;
      }

      return this.target.getBoolean(key);
    } catch (JSONException e) {
      return defaultValue;
    }
  }

  @Override
  public int getForegroundColor() {
    return super.getForegroundColor("#272937");
  }

  public boolean getDisableAdditionalReviewScreen() {
    return this.getBoolean("isDisableAdditionalReviewScreen", false);
  }

  public boolean getEnableAdditionalReviewTag() {
    return this.getBoolean("isEnableAdditionalReviewTag", true);
  }

  public int getTagImageColor() {
    return this.color.getColor(this.target, "tagImageColor", "#CC0044");
  }

  public int getTagTextColor() {
    return this.color.getColor(this.target, "tagTextColor", "#272937");
  }

  public int getReviewImage() {
    return this.image.getSource("reviewImage");
  }

  public int getTagImage() {
    return this.image.getSource("tagImage");
  }
}

File: ios/Theme/AdditionalReview.swift
public class AdditionalReview: ViewStyle {
  private static let KEY: String = "additionalReview"
  private let style: Style
  private let target: NSDictionary?
  private let color: Color
  private let image: Image

  init(target: NSDictionary?) {
    self.style = Style()

    self.target = self.style.getTarget(target, key: AdditionalReview.KEY)
    self.color = Color()
    self.image = Image(target: self.target)

    super.init(target: self.target, key: AdditionalReview.KEY)
  }
  
  private func getBoolean(_ key: String, defaultValue: Bool) -> Bool {
    if !self.style.exists(self.target, key: key) {
      return defaultValue
    }

    return self.target?[key] as! Bool
  }
  
  override func getForegroundColor() -> UIColor {
    return super.getForegroundColor(defaultColor: "#272937")
  }
  
  public func getDisableAdditionalReviewScreen() -> Bool {
    return self.getBoolean("isDisableAdditionalReviewScreen", defaultValue: false)
  }
  
  public func getEnableAdditionalReviewTag() -> Bool {
    return self.getBoolean("isEnableAdditionalReviewTag", defaultValue: true)
  }
  
  public func getDisplayTime() -> Double {
    let defaultDisplayTime: Double = 2.0
    let key = "displayTime"
      
    if !self.style.exists(self.target, key: key) {
      return defaultDisplayTime
    }
      
    let displayTime = self.target?[key] as! Double
    return displayTime < 0 ? defaultDisplayTime : displayTime
  }
  
  public func getTagImageColor() -> UIColor {
    return self.color.getColor(self.target, key: "tagImageColor", defaultColor: "#CC0044")
  }
  
  public func getTagTextColor() -> UIColor {
    return self.color.getColor(self.target, key: "tagTextColor", defaultColor: "#272937")
  }
  
  public func getReviewImage() -> UIImage? {
    return self.image.getSource("reviewImage")
  }
  
  public func getTagImage() -> UIImage? {
    return self.image.getSource("tagImage")
  }
}
